src/iGenVar.cpp

```diff
@@ -1,72 +1,9 @@
-                                                  {"simple_clustering",
-                                                   clustering_methods::simple_clustering},
-                                                  {"1", clustering_methods::hierarchical_clustering},
-                                                  {"hierarchical_clustering",
-                                                   clustering_methods::hierarchical_clustering},
-                                                  {"2", clustering_methods::self_balancing_binary_tree},
-                                                  {"self_balancing_binary_tree",
-                                                   clustering_methods::self_balancing_binary_tree},
-                                                  {"3", clustering_methods::candidate_selection_based_on_voting},
-                                                  {"candidate_selection_based_on_voting",
-                                                   clustering_methods::candidate_selection_based_on_voting}};
-};
-
-// Specialise a mapping from an identifying string to the respective value of your type refinement_methods. With the
-// help of this function, you're able to call ./detect_breackends with -r 0 and -r no_refinement and get the same
-// result.
-auto enumeration_names(refinement_methods)
-{
-    return std::unordered_map<std::string, refinement_methods>{{"0", refinement_methods::no_refinement},
-                                                               {"no_refinement",
-                                                                refinement_methods::no_refinement},
-                                                               {"1", refinement_methods::sViper_refinement_method},
-                                                               {"sViper_refinement_method",
-                                                                refinement_methods::sViper_refinement_method},
-                                                               {"2", refinement_methods::sVirl_refinement_method},
-                                                               {"sVirl_refinement_method",
-                                                                refinement_methods::sVirl_refinement_method}};
-};
-
-struct cmd_arguments
-{
-    std::filesystem::path alignment_short_reads_file_path{""};
-    std::filesystem::path alignment_long_reads_file_path{""};
-    std::filesystem::path insertion_file_path{};
-    std::filesystem::path output_file_path{};
-    std::vector<detection_methods> methods{cigar_string, split_read, read_pairs, read_depth};   // default: all methods
-    clustering_methods clustering_method{simple_clustering};                                    // default: simple clustering method
-    refinement_methods refinement_method{no_refinement};                                        // default: no refinement
-    uint64_t min_var_length = 30;
-};
```

Comment:
Why do you move the `cmd_arguments` struct to `variant_detection.hpp`? Wasn't it quite logical to have it close to the argument parsing method?

Reply:
I will move it back in the next PR (#104) it was necessary as `cmd_arguments` is not known in `variant_detection.hpp`, but in the next PR I'll move the whole function into iGenVar.cpp

Reply:
I added a TODO for this.